things/postgres/channels.go

```diff
@@ -34,9 +35,13 @@ func NewChannelRepository(db *sqlx.DB) things.ChannelRepository {
 	}
 }
 
-func (cr channelRepository) Save(_ context.Context, channel things.Channel) (string, error) {
+func (cr channelRepository) Save(ctx context.Context, channel things.Channel) (string, error) {
+	span := opentracing.SpanFromContext(ctx)
+	defer span.Finish()
+
 	q := `INSERT INTO channels (id, owner, name, metadata)
-        VALUES (:id, :owner, :name, :metadata);`
+		VALUES (:id, :owner, :name, :metadata);`
+	span.SetTag("sql.statement", q)
```

Comment:
I suggest that you create interface for the db and write a middleware for that interface that will add these tags. Idea is to not mix repository code with an open tracing code. So if you create this middleware, when you call `NamedExec`, it will under the hood call your middleware, that will add tags, and will call real `NamedExec`. For an example take a look at our logging middlewares, and middlewares in `things/tracing` package.

Reply:
Interface added.